powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileTokenOfflineController.java

```diff
@@ -85,20 +90,23 @@
     private final AuthMethodQueryService authMethodQueryService;
     private final WebFlowServicesConfiguration webFlowServicesConfiguration;
     private final PushMessageService pushMessageService;
+    private final HttpSession httpSession;
```

Comment:
Can this be autowired? Isn't the controller a singleton?
I would expect to have it as parameter similar to request.

Reply:
Yes, Spring uses a proxy for `HttpSession`.